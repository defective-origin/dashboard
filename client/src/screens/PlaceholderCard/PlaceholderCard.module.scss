@import 'src/theme';

:root {
  // --ui-framework-color: var(--account-framework-color);
}

.PlaceholderCard {
  display: flex;
  align-items: center;
  justify-content: center;
}

.Message {
  white-space: wrap;
  text-align: center;
  text-decoration: none;
}
