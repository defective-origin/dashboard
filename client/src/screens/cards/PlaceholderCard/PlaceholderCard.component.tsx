import React from 'react'

// ---| core |---
import { cn } from 'tools'
import { useLocale } from 'locale'

// ---| pages |---
// ---| screens |---
// ---| components |---
import Item, { ItemProps } from 'components/Item'
import Text from 'components/Text'

// ---| self |---
import css from './PlaceholderCard.module.scss'

export type PlaceholderCardProps = ItemProps & {
  name?: string
  className?: string
  children?: React.ReactNode
}

/**
 * Component description.
 *
 * How to use
 * @example
 * <PlaceholderCard />
 */
export function PlaceholderCard(props: PlaceholderCardProps): JSX.Element {
  const { name = 'COMPONENT', children, className, ...otherProps } = props
  const _className = cn(css.PlaceholderCard, className)
  const locale = useLocale()

  return (
    <Item className={_className} p='xl' {...otherProps}>
      <Text.Body2
        className={css.Message}
        color='secondary'
        format='capitalize'
        content={locale.t('MESSAGES.UNDER_CONSTRUCTION', { name })}
      />

      {children}
    </Item>
  )
}

PlaceholderCard.displayName = 'PlaceholderCard'

export default PlaceholderCard
