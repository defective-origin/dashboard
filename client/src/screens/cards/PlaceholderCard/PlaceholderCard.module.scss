@import 'src/theme';

:root {
  // --ui-framework-color: var(--account-framework-color);
}

.PlaceholderCard {
  display: flex;
  align-items: center;
  justify-content: center;
  border: var(--border);
  box-shadow: var(--box-shadow);
  border-radius: var(--border-radius);
  background: var(--card-background);
}

.Message {
  white-space: wrap;
  text-align: center;
  text-decoration: none;
}
