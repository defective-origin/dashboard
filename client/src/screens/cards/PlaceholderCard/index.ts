/* INJECT_IMPORT_PLACE */

/* INJECT_DEFAULT_EXPORT_PLACE */
export { default } from './PlaceholderCard.component'

/* INJECT_EXPORT_PLACE */
export * from './PlaceholderCard.component'
