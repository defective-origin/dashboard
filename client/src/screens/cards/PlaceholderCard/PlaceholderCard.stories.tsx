// eslint-disable-next-line no-restricted-imports
import { field, params } from '../../../../.storybook/tool'
import type { Meta, StoryObj } from '@storybook/react'
import PlaceholderCard from './PlaceholderCard.component'

const meta: Meta<typeof PlaceholderCard> = {
  title: 'Screens/CARDS/PlaceholderCard',
  component: PlaceholderCard,
  tags: ['autodocs'],
  argTypes: {
    name: field.string(),
    className: field.string(),
    children: field.reactNode(true),
  },
}

export default meta

type Story = StoryObj<typeof PlaceholderCard>

export const Demo: Story = {
  parameters: params('PlaceholderCard'),
  args: {
    name: 'COMPONENT',
  },
}
