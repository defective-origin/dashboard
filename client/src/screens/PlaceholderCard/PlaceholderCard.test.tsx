// ---| tests |---
import { render } from 'tests'

// ---| self |---
import PlaceholderCard from './PlaceholderCard.component'
// import usePlaceholderCard from './PlaceholderCard.hook'
// import PlaceholderCardProvider, PlaceholderCard from './PlaceholderCard.context'

describe('[PlaceholderCard] component', () => {
  it('should render component', () => {
    const container = render(<PlaceholderCard />)

    expect(container.snapshot()).toMatchSnapshot()
  })
})

// describe('[PlaceholderCard] hook', () => {
//   it('should return options', () => {
//     const { result } = renderHook(() => usePlaceholderCard())
//
//     expect(result.current).toEqual(null)
//   })
// })

// describe('[PlaceholderCard] context', () => {
//   const wrapper = (props: PlaceholderCardProps) => <PlaceholderCardProvider { ...props } />
// 
//   it('should return options', () => {
//     const { result } = renderHook(() => usePlaceholderCard(), { wrapper })
//
//     expect(result.current).toBeTruthy()
//   })
// })
