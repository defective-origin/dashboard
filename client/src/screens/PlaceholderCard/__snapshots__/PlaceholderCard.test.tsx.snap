// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`[PlaceholderCard] component > should render component 1`] = `
<DocumentFragment>
  <div
    class="item p--xl _PlaceholderCard_dbf372"
  >
    <p
      class="MuiTypography-root MuiTypography-body2 MuiTypography-alignLeft text text--sm _Message_dbf372 css-zvssqi-MuiTypography-root"
    >
      Component Is Under Construction.
    </p>
  </div>
</DocumentFragment>
`;
